common/src/main/java/dev/latvian/mods/itemfilters/item/StrongNBTFilterItem.java

```diff
@@ -113,4 +116,15 @@ public boolean filter(ItemStack filter, ItemStack stack) {
 		CompoundTag tag2 = stack.getTag();
 		return Objects.equals(tag1, tag2);
 	}
+
+	@Override
+	public String getHelpKey() {
+		return "itemfilters.help_text.nbt";
+	}
+
+    @Override
+    public void getItems(ItemStack filter, Set<Item> set) {
+		// any item could potentially have NBT, so we need the lot
+		Registry.ITEM.forEach(set::add);
+    }
```

Comment:
Now that I think about it I'm... kind of worried about the possible performance impact this could have, which is probably why Lat never added this in the first place. Would you say it's still reasonable to use this with arbitrary large registry sizes?

Reply:
Disregard this, I had another look at the rest of the mod (after... about a year of not wanting to touch this) and... wow these default implementations are woefully inefficient. I somewhat understand why, but still, this entire mod might need to be redesigned at some point haha